src/vs/workbench/contrib/terminal/browser/terminalEditor.ts

```diff
@@ -41,10 +43,19 @@ export class TerminalEditor extends EditorPane {
 			this.layout(this._lastDimension);
 		}
 		this._editorInput.terminalInstance?.setVisible(true);
+		if (this._editorInput.terminalInstance) {
+			// since the editor does not monitor focus changes, for ex. between the terminal
+			// panel and the editors, this is needed so that the active instance gets set
+			// when focus changes between them.
+			this._register(this._editorInput.terminalInstance!.onFocused(() => this.focus()));
+		}
 	}
 
 	override focus() {
-		this._editorInput?.terminalInstance?.focus();
+		if (!this._editorInput?.terminalInstance) {
+			return;
+		}
+		this._terminalEditorService.setActiveInstance(this._editorInput.terminalInstance);
```

Comment:
I think we still need the call to `focus`, this fixed the issue where you focus the editor by clicking the tab and start typing into the terminal.